client/state/signup/progress/reducer.js

```diff
@@ -116,6 +117,9 @@ export default withSchemaValidation( schema, ( state = {}, action ) => {
 			return saveStep( state, action );
 		case SIGNUP_PROGRESS_SUBMIT_STEP:
 			return submitStep( state, action );
+		case SIGNUP_STEPS_SITE_TYPE_SET:
+			delete state[ 'domains-with-preview' ];
```

Comment:
Can't say I like this, especially using a hardcoded string, but seems it's the "signup" way. The main thing I'm concerned is that this is probably not the only flow that free subdomain suggestions show up. But how to programmatically get the rest?